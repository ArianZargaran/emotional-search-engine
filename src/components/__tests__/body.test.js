import React from "react";
import Body from "../body";
import MainHeader from "../main-header";
import SearchEngine from "../search-engine/search-engine";
import ResultsSection from "../results-section/results-section";

const setup = (extraProps) => {
  const component = shallow(<Body />)
  const props = {
    ...extraProps
  };

  return { component, props }
};

describe("Body", () => {
  it('renders Body correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<Body />);
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });

  it("renders right components in the right order", () => {
    const { component } = setup();
    const instanceOrder = (idx, instance) => component.find(".body-wrapper").childAt(idx).is(instance);
 
    expect(instanceOrder(0, MainHeader)).toBeTruthy();
    expect(instanceOrder(1, SearchEngine)).toBeTruthy();
    expect(instanceOrder(2, ResultsSection)).toBeTruthy();
  });
});