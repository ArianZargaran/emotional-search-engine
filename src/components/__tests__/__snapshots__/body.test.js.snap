// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Body correctly 1`] = `
<div>
  <UNDEFINED>
    Emotional Search Engine
  </UNDEFINED>
  <searchEngine />
  <ResultsSection />
</div>
`;
